### ISSUE TITLE
Consider Removing New Line Characters when Comparing Computed Constructs

### ISSUE THREAD

**Issue by james-menzies** [sqlalchemy/alembic#1121]
I'm working with a particularly long Computed Column construct, which is really long and therefore contains line breaks for readability.

When this construct gets imported into the underlying database (Postgres in my case), the construct gets reformatted with line breaks inserted in different places.

This creates a small side-effect, where alembic now thinks that the original computed clause provided by the migration file is *different* to the one in the database, and therefore emits a warning.
```
/usr/local/lib/python3.10/site-packages/alembic/autogenerate/compare.py:1090: 
UserWarning: Computed default on ***  cannot be modified
util.warn("Computed default on %s.%s cannot be modified" % (tname, cname))
```

After inspecting the code, I found this method, which appears to sanitize the database computed clause as well as the local one:

https://github.com/sqlalchemy/alembic/blob/fc451e60731ad5b8d73255f0a1996e1d4e0f20de/alembic/autogenerate/compare.py#L1047-L1055

I feel that we should include line breaks in the list of tokens that should be removed before a comparison takes place, but happy to here what thoughts you have on this.

Thanks!

**Comment by zzzeek** [comment]
yes we should do this, computed strings with newlines in them had not been considered.

**Comment by CaselIT** [comment]
It could trigger false negative in case the computed expression is a string with a new-line that was changed, but it's probably a vary rare case, so it makes sense to remove them from comparison.

**Comment by IMGillusion** [comment]
This looks like it's already resolved on `main` — the change requested here (stripping newlines/tabs before comparing computed expressions) landed in 06aee70e (PR #1393, 2024-01-10), courtesy of Georg Wicke-Arndt:

    -    return re.sub(r"[ \(\'\"`\[\]]", "", sqltext).lower()
    +    return re.sub(r"[ \(\'\"`\[\]\t\r\n]", "", sqltext).lower()

along with a regression test combo:

    lambda: (sa.Computed("bar*5"), sa.Computed("bar * \r\n\t5")),

So moving/inserting line breaks inside a Computed expression no longer trips the "Computed default on ... cannot be modified" warning — exactly the side-effect reported here.

This issue looks like the original report that #1391 / #1393 were fixed against, so it can probably be closed as resolved.
